sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java

```diff
@@ -20,21 +25,70 @@
      */
     XML;
 
+    private static final ClientLogger LOGGER = new ClientLogger(SerializerEncoding.class);
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final Map<String, SerializerEncoding> SUPPORTED_MIME_TYPES;
+    private static final TreeMap<String, SerializerEncoding> SUPPORTED_SUFFIXES;
```

Comment:
Watch out, this could hit transient concurrent modification errors. Mind trying out `ConcurrentSkipListMap`?

Reply:
We never modify this list other than at static construction.